taroscript/send.go

```diff
@@ -254,6 +254,105 @@ func SignVirtualTransaction(input *taropsbt.VInput, outputs []*taropsbt.VOutput,
 	return nil
 }
 
+// CreateOutputCommitments creates the final set of TaroCommitments representing
+// the asset send. The input TaroCommitment must be set.
+func CreateOutputCommitments(inputCommitment *commitment.TaroCommitment,
+	input *taropsbt.VInput,
+	outputs []*taropsbt.VOutput) ([]*commitment.TaroCommitment, error) {
+
+	inputAsset := input.Asset
+
+	// Remove the spent Asset from the AssetCommitment of the sender. Fail
+	// if the input AssetCommitment or Asset were not in the input
+	// TaroCommitment.
+	changeTaroCommitment, err := inputCommitment.Copy()
```

Comment:
Important to mind all the copies in the old code, as these are in place to avoid unwanted mutations.

Reply:
I think I did them justice and carefully adapted all of the calls to `Copy()`. But if you spot one I missed, please let me know!